apps/mobile/src/features/Settings/components/Currency/CurrencySection/CurrencySection.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { View, Text, YStack } from 'tamagui'
+import { CurrencyItem } from '../CurrencyItem'
+import { CURRENCY_DATA } from '@/src/config/currencies'
+import type { CurrencySectionProps } from '../Currency.types'
+
+export const CurrencySection: React.FC<CurrencySectionProps> = ({
+  title,
+  currencies,
+  selectedCurrency,
+  onCurrencySelect,
+}) => (
+  <YStack marginBottom="$4">
+    <View paddingVertical="$2">
+      <Text fontSize="$4" fontWeight="500" color="$colorSecondary">
+        {title}
+      </Text>
+    </View>
+    <YStack gap="$4">
+      {currencies.map((currency) => {
+        const currencyInfo = CURRENCY_DATA[currency.toUpperCase()]
+        if (!currencyInfo) {
+          return null
+        }
+
+        return (
+          <CurrencyItem
+            key={currency}
+            code={currency.toUpperCase()}
+            symbol={currencyInfo.symbol}
+            name={currencyInfo.name}
+            isSelected={selectedCurrency.toUpperCase() === currency.toUpperCase()}
+            onPress={() => onCurrencySelect(currency.toLowerCase())}
+          />
+        )
+      })}
+    </YStack>
+  </YStack>
+)
```

Comment:
could we please move the business logic out of the presentational layer?

```suggestion
export const CurrencySection: React.FC<CurrencySectionProps> = ({
  title,
  currencies,
  selectedCurrency,
  onCurrencySelect,
}) => {
const filteredCurrencies = currencies.filter(currency => CURRENCY_DATA[currency.toUpperCase()])

return (
  <YStack marginBottom="$4">
    <View paddingVertical="$2">
      <Text fontSize="$4" fontWeight="500" color="$colorSecondary">
        {title}
      </Text>
    </View>
    <YStack gap="$4">
      {filteredCurrencies.map((currency) =>  (
          <CurrencyItem
            key={currency}
            code={currency.toUpperCase()}
            symbol={currencyInfo.symbol}
            name={currencyInfo.name}
            isSelected={selectedCurrency.toUpperCase() === currency.toUpperCase()}
            onPress={() => onCurrencySelect(currency.toLowerCase())}
          />
        ))}
    </YStack>
  </YStack>
)
}
```